client/src/components/Picture/UploadPic.js

```diff
@@ -0,0 +1,237 @@
+      setUploadError(
+          t("error.avatar.fileSizeTooLarge")
+      );
+    } else {
+      const fileReader = new FileReader();
+      fileReader.onloadend = () => {
+        const image = new Image();
+        image.src = fileReader.result;
+        image.onload = () => {
+          if (image.height < 250 || image.width < 250) {
+            setUploadError(
+                t("error.avatar.imageDimensionTooSmall")
+            );
+          } else if (image.height / image.width < 0.5) {
+            setUploadError(
+              t("error.avatar.imageDimensionTooWide")
+            );
+          }
+        };
+        setPhotoURL(fileReader.result);
+      };
+      if (file) {
+        fileReader.readAsDataURL(file);
+      }
+    }
+    setModalVisible(true);
+  };
+
+  const savePhoto = async () => {
+    const canvas = document.createElement("canvas");
+    const scaleX = image.naturalWidth / image.width;
+    const scaleY = image.naturalHeight / image.height;
+    canvas.width = crop.width;
+    canvas.height = crop.height;
+    const ctx = canvas.getContext("2d");
+    ctx.drawImage(
+      image,
+      crop.x * scaleX,
+      crop.y * scaleY,
+      crop.width * scaleX,
+      crop.height * scaleY,
+      0,
+      0,
+      crop.width,
+      crop.height,
+    );
+    const reader = new FileReader();
+    canvas.toBlob((blob) => {
+      let uploadResponse;
+      let endPoint = "/api/users/current/avatar";
+      const formData = new FormData();
+      formData.append("file", blob);
+      if (user.ownerId) {
+        endPoint = `/api/organisations/${user._id}/avatar`;
+      }
+      reader.readAsDataURL(blob);
+      reader.onloadend = async () => {
+        try {
+          uploadResponse = await axios({
+            method: "post",
+            url: endPoint,
+            data: formData,
+            headers: {
+              accept: "application/json",
+              "Content-Type": `multipart/form-data`,
+            },
+          });
+          if (uploadResponse.status == 200) {
+            setModalVisible(false);
+            window.location.reload(false);
+            console.log("Success");
+          }
+        } catch (error) {
+          setUploadError("error.avatar.networkError");
+          console.log({
+            error,
+          });
+        }
+      };
+    });
+    setModalVisible(false);
+  };
+
+  const closeModal = () => {
+    imgUpload.current.value = "";
+    setUploadError("");
+    setModalVisible(false);
+  };
+
+  const retry = () => {
+    imgUpload.current.value = "";
+    imgUpload.current.click();
+  };
+
+  const cropModal = () => {
+    return (
+      <Modal
+        visible={modalVisible}
+        okText={"Save"}
+        onOk={savePhoto}
+        destroyOnClose={true}
+        closable={false}
+        onCancel={() => setModalVisible(false)}
+        maskClosable={false}
+        footer={[
+          <CustomCancelButton key="cancel" onClick={closeModal}>
+            Cancel
```

Comment:
Ah, sorry I should have reviewed first before approving. Button text should be localized as well.

Reply:
No problem, will make these changes tommorow.